### PR TITLE
report semgrep findings and fix findings assertions

### PR DESCRIPTION
**Pull request by clavedeluna**
I initially thought asserting findings is not None was sufficient, but in cases when the result of matching findings is `[]` we also want to be aware.

I updated some codemods where it was easy to do the matching, but created https://github.com/pixee/codemodder-python/issues/738 for that codemod

This fix also brought to light that we weren't reporting semgrep findings so fixed that too

### CODE REVIEW

File: src/codemodder/result.py [hunk 1]
```diff
@@ -68,18 +68,39 @@ def match_location(self, pos: CodeRange, node: cst.CSTNode) -> bool:
...
             locations=cls.extract_locations(sarif_result),
             codeflows=cls.extract_code_flows(sarif_result),
             related_locations=cls.extract_related_locations(sarif_result),
+            finding_id=rule_id,
+            finding=Finding(
+                id=rule_id,
+                rule=Rule(
+                    id=rule_id,
+                    name=rule_id,
+                    url=semgrep_url_from_id(rule_id),
```
Comment (drdavella) [inline src/codemodder/result.py]: This is specific to Semgrep but it looks like this is being added to a more general `SarifResult` class?

**Review by drdavella — CHANGES_REQUESTED** [review]

File: src/codemodder/result.py (hunk 1 above)
Comment (clavedeluna) [reply src/codemodder/result.py]: yeah do you think this is wrong? while it's a generic sarif result, wouldn't all sarif results have to have findings too?

**Review by andrecsilva — APPROVED** [review]

File: src/codemodder/result.py (hunk 1 above)
Comment (drdavella) [reply src/codemodder/result.py]: My concern is specifically about the call to `semgrep_url_from_id`, which only applies to Semgrep results, whereas this class appears to be used for multiple SARIF-based tools.

Comment (clavedeluna) [reply src/codemodder/result.py]: wooops Im an idiot, of course! Good review Dan, will update

**Review by drdavella — APPROVED** [review]

**Pull request merged**